webviz_subsurface/plugins/_pvt_plot.py

```diff
@@ -0,0 +1,640 @@
+
+import pandas as pd
+import dash_html_components as html
+from dash.dependencies import Input, Output, State
+from dash.exceptions import PreventUpdate
+import dash_core_components as dcc
+import webviz_core_components as wcc
+from webviz_config.common_cache import CACHE
+from webviz_config import WebvizPluginABC
+
+from .._datainput.pvt_data import load_pvt_dataframe
+from .._datainput.fmu_input import load_csv
+
+
+class PvtPlot(WebvizPluginABC):
+    """Visualizes formation volume factor and viscosity data \
+for oil, gas and water from both Eclipse **init** and **include** files.
+
+---
+
+* **`ensembles`:** Which ensembles in `shared_settings` to visualize.
+* **`pvt_relative_file_path`:** Local path to a csv file in each \
+    realization with dumped pvt data.
+* **`read_from_init_file`:** A boolean flag stating if data shall be \
+    read from an Eclipse INIT file instead of an INCLUDE file. \
+    This is only used when **pvt_relative_file_path** is not given.
+
+---
+The minimum requirement is to define `ensembles`.
+
+If no `pvt_relative_file_path` is given, the PVT data will be extracted automatically
+from the simulation decks of individual realizations using `fmu_ensemble` and `ecl2data_frame`.
+If the `read_from_init_file` flag is set to True, then the extraction procedure in
+`ecl2data_frame` will be replaced by an individual extracting procedure that reads the
+normalized Eclipse INIT file.
+Note that the latter two extraction methods can be very slow for larger data and are therefore
+not recommended unless you have a very simple model/data deck.
+
+`pvt_relative_file_path` is a path to a file stored per realization (e.g. in \
+`share/results/tables/pvt.csv`). `pvt_relative_file_path` columns:
+* One column named `KEYWORD` or `TYPE`: with Flow/Eclipse style keywords
+    (e.g. `PVTO` and `PVDG`).
+* One column named `PVTNUM` with integer `PVTNUM` regions.
+* One column named `GOR` or `RS` with the gas-oil-ratio as the primary variate.
+* One column named `PRESSURE` with the fluids pressure as the secondary variate.
+* One column named `VOLUMEFACTOR` as the first covariate.
+* One column named `VISCOSITY` as the second covariate.
+
+The file can e.g. be dumped to disc per realization by a forward model in ERT using
+`ecl2data_frame` or `pyscal`.
+"""
+
+    PHASES = {"OIL": "PVTO", "GAS": "PVDG", "WATER": "PVTW"}
+
+    def __init__(
+        self,
+        app,
+        ensembles: List[str],
+        pvt_relative_file_path: str = None,
+        read_from_init_file: bool = False,
+    ):
+
+        super().__init__()
+
+        self.ensemble_paths = {
+            ensemble: app.webviz_settings["shared_settings"]["scratch_ensembles"][
+                ensemble
+            ]
+            for ensemble in ensembles
+        }
+
+        self.plotly_theme = app.webviz_settings["theme"].plotly_theme
+
+        self.pvt_relative_file_path = pvt_relative_file_path
+
+        if self.pvt_relative_file_path is not None:
```

Comment:
could be nicer to switch the order of the if-statement to get `if self.pvt_relative_file_path is None` instead, but that is just a preference :wink:

Reply:
Agree, it somehow reads nicer the other way around ;)